packages/react/src/components/Toggle/next/Toggle.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright IBM Corp. 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { settings } from 'carbon-components';
+import { useControllableState } from '../../../internal/useControllableState';
+
+const { prefix } = settings;
+
+export function Toggle({
+  className,
+  defaultToggled = false,
+  disabled = false,
+  hideLabel = false,
+  id,
+  labelA = 'Off',
+  labelB = 'On',
+  labelText,
+  onClick = () => {},
+  onToggle,
+  size = 'md',
+  toggled,
+}) {
+  const [checked, setChecked] = useControllableState(
+    toggled,
+    onToggle,
+    defaultToggled
+  );
+
+  function handleClick(e) {
+    setChecked(!checked);
+    onClick(e);
+  }
+
+  const isSm = size === 'sm';
+  const sideLabel = hideLabel ? labelText : checked ? labelB : labelA;
+
+  const wrapperClasses = classNames(
+    `${prefix}--toggle`,
+    {
+      [`${prefix}--toggle--disabled`]: disabled,
+    },
+    className
+  );
+
+  const labelTextClasses = classNames(`${prefix}--toggle__label-text`, {
+    [`${prefix}--visually-hidden`]: hideLabel,
+  });
+
+  const appearanceClasses = classNames(`${prefix}--toggle__appearance`, {
+    [`${prefix}--toggle__appearance--sm`]: isSm,
+  });
+
+  const switchClasses = classNames(`${prefix}--toggle__switch`, {
+    [`${prefix}--toggle__switch--checked`]: checked,
+  });
+
+  return (
+    <div className={wrapperClasses}>
+      <button
+        id={id}
+        tabIndex={0}
```

Comment:
Does this need an explicit `tabIndex` so that it can still be focused when disabled?

Reply:
Nope, I think that was an artifact from the refactoring process. It's removed now.